src/source/src/manager.rs

```diff
@@ -130,27 +132,53 @@ pub struct MemSourceManager {
 #[async_trait]
 impl SourceManager for MemSourceManager {
     fn get_source(&self, table_id: &TableId) -> Result<SourceDesc> {
-        let sources = self.get_sources()?;
-        sources.get(table_id).cloned().ok_or_else(|| {
+        let inner = self.inner.lock();
+        inner.sources.get(table_id).cloned().ok_or_else(|| {
             InternalError(format!("Get source table id not exists: {:?}", table_id)).into()
         })
     }
 
-    fn drop_source(&self, table_id: &TableId) -> Result<()> {
-        let mut sources = self.get_sources()?;
-        ensure!(
-            sources.contains_key(table_id),
-            "Source does not exist: {:?}",
-            table_id
-        );
-        sources.remove(table_id);
-        Ok(())
+    fn insert_table_source(
+        &self,
+        table_id: &TableId,
+        info: &TableSourceInfo,
+    ) -> Result<SourceDesc> {
+        let mut inner = self.inner.lock();
+        let actor_num = inner.source_actor_num.entry(*table_id).or_insert(0usize);
+        *actor_num += 1;
```

Comment:
Maybe we can also use `Arc` to manage the ref count automatically, by storing a weak reference in the map and lazily cleaning it up.

So that there's also no need to record the mapping in the stream manager: when the source executor for the table is dropped, the ref count can also be decremented.

Reply:
Is there such a case that when we are doing drop_actors/force_stop_actors, bug the other SourceRef replica in source executors is still not dropped? I am not very sure.. @yezizp2012 @BugenZhao

Reply:
But if JoinHandle.abort() has no guarantee to drop the strong reference in source executor immediately, there may be a case that when we doing drop_actors for some aborted actors, their responding table source's strong reference count doesn't show 0, and we believe some executor is still using it, and we keep it. Then when will such source desc be removed from local source manager? : (
We will only simply clear the source map totally in doing force_stop_actors but not in doing drop_actors..

Reply:
In normal scenario, `drop_actors` will be called after actor exit normally. And I believe the reference will be dropped. And for force_stop_actors, if we really are not sure about this, I think we can add a cleaning logic for source map inside to guarantee the drop.

Reply:
Add a clear_all_sources during drop_all_actors according to @yezizp2012 's suggestion. At most time the abort() will guarantee that strong references are dropped before try_drop_source, so it may be ok. : ) We can still use Weak/Arc to count references.